src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -444,12 +444,18 @@ private IntPtr AsUserDefined(in Guid riid)
 
             private void SetFlag(CreateComInterfaceFlagsEx flag)
             {
-                Interlocked.Or(ref Flags, flag);
+                // Using Interlocked.Or<T> will trigger type checks
+                // that can cause deadlocks when running during a GC.
```

Comment:
```suggestion
                // Interlocked.Or<T> cannot be used here. It would trigger type checks that can cause 
                // deadlocks when called during a GC by NativeAOT TrackerObjectManager.
```
More details in the comment?